compiler/src/dotty/tools/dotc/tastyreflect/ReflectionCompilerInterface.scala

```diff
@@ -1506,148 +1516,84 @@ class ReflectionCompilerInterface(val rootContext: core.Contexts.Context) extend
+  def Symbol_method(self: Symbol)(name: String)(given Context): List[Symbol] =
     appliedTypeRef(self).allMembers.iterator.map(_.symbol).collect {
       case sym if isMethod(sym) && sym.name.toString == name => sym.asTerm
     }.toList
 
-  def ClassDefSymbol_methods(self: Symbol)(given Context): List[DefDefSymbol] =
+  def Symbol_methods(self: Symbol)(given Context): List[Symbol] =
     appliedTypeRef(self).allMembers.iterator.map(_.symbol).collect {
       case sym if isMethod(sym) => sym.asTerm
     }.toList
 
   private def isMethod(sym: Symbol)(given Context): Boolean =
     sym.isTerm && sym.is(Flags.Method) && !sym.isConstructor
 
-  def ClassDefSymbol_caseFields(self: Symbol)(given Context): List[ValDefSymbol] =
+  def Symbol_caseFields(self: Symbol)(given Context): List[Symbol] =
     if (!self.isClass) Nil
     else self.asClass.paramAccessors.collect {
       case sym if sym.is(Flags.CaseAccessor) => sym.asTerm
     }
 
-  def ClassDefSymbol_companionClass(self: Symbol)(given Context): Option[ClassDefSymbol] = {
-    val sym = self.companionModule.companionClass
-    if (sym.exists) Some(sym.asClass) else None
-  }
-
-  def ClassDefSymbol_companionModule(self: Symbol)(given Context): Option[ValDefSymbol] = {
-    val sym = self.companionModule
-    if (sym.exists) Some(sym.asTerm) else None
-  }
-
-  def ClassDefSymbol_moduleClass(self: Symbol)(given Context): Option[Symbol] = {
-    val sym = self.moduleClass
-    if (sym.exists) Some(sym.asTerm) else None
-  }
-
   private def isField(sym: Symbol)(given Context): Boolean = sym.isTerm && !sym.is(Flags.Method)
 
-  def ClassDefSymbol_of(fullName: String)(given ctx: Context): ClassDefSymbol = ctx.requiredClass(fullName)
+  def Symbol_of(fullName: String)(given ctx: Context): Symbol =
+    ctx.requiredClass(fullName)
 
-  type TypeDefSymbol = core.Symbols.TypeSymbol
-
-  def matchTypeDefSymbol(symbol: Symbol)(given Context): Option[TypeDefSymbol] =
-    if (symbol.isType) Some(symbol.asType) else None
-
-  def TypeDefSymbol_tree(self: TypeDefSymbol)(given Context): TypeDef =
-    FromSymbol.typeDefFromSym(self)
-  def TypeDefSymbol_isTypeParam(self: TypeDefSymbol)(given Context): Boolean =
+  def Symbol_isTypeParam(self: Symbol)(given Context): Boolean =
     self.isTypeParam
 
-  type TypeBindSymbol = core.Symbols.TypeSymbol
-
-  def matchTypeBindSymbol(symbol: Symbol)(given Context): Option[TypeBindSymbol] =
-    if (symbol.isType && symbol.is(core.Flags.Case)) Some(symbol.asType) else None
-
-  def TypeBindSymbol_tree(self: TypeBindSymbol)(given Context): TypeBind =
-    FromSymbol.typeBindFromSym(self)
-
-  type TermSymbol = core.Symbols.TermSymbol
-
-  def matchTermSymbol(symbol: Symbol)(given Context): Option[TermSymbol] =
-    if (symbol.isTerm) Some(symbol.asTerm) else None
-
-  type DefDefSymbol = core.Symbols.TermSymbol
+  def Symbol_isTypeBindSymbol(symbol: Symbol)(given Context): Boolean =
+    symbol.isType && symbol.is(core.Flags.Case)
```

Comment:
`isTypeBindSymbol` -> `isPatternBoundType`?

A case class symbol will pass the check.